packages/medusa/src/services/order-edit.ts

```diff
@@ -678,6 +691,23 @@ export default class OrderEditService extends TransactionBaseService {
       orderEdit.requested_at = new Date()
       orderEdit.requested_by = context.loggedInUserId
 
+      const total = await this.getTotals(orderEdit.id)
+
+      if (total.difference_due > 0) {
+        const paymentCollection = await this.paymentCollectionService_
+          .withTransaction(manager)
+          .create({
+            type: PaymentCollectionType.ORDER_EDIT,
+            amount: total.difference_due,
+            currency_code: orderEdit.order.currency_code,
+            region_id: orderEdit.order.region_id,
+            description: description,
+            created_by: context.loggedInUserId as string,
+          })
+
+        orderEdit.payment_collection_id = paymentCollection.id
+      }
```

Comment:
thought: I think we should avoid coupling the payment collection with the order edit and instead allow them to be logically separate. E.g., you can create an order edit without always having to create a payment collection if you are using the services directly (i.e. not through the API).

In the core API we will create the PaymentCollection for you though - i.e. move this to `create-order-edit`

Reply:
In the standard case that will happen yes, which is why I would move this logic to the API handler: 

```ts
// packages/medusa/src/api/routes/admin/order-edits/create-order-edit.ts

export default async (req, res) => {
  // ...

  const createdOrderEdit = await orderEditService.create(data)
  
  if (createdOrderEdit.difference_due > 0) {
    // create payment collection
    await orderEditService.update(createdOrderEdit.id, { payment_collection_id: payCol.id })
  }

  // ...
}
```

My thinking is that if you want to use the OrderEdit capabilities without enforcing the payment collection you can do so when calling the OrderEditService directly, but then it is your responsibility to figure out what to do with payments. Ultimately this gives developers more powers in OMS customizations.

Reply:
The main point here is that the default API would use our building blocks to create a business logic where a payment collection is always associated with an orderEdit that has positive difference_due, but if you need customizations to your OMS setup we have decoupled those two ideas enough for you to not have to "hack" around the payment collection.

Reply:
If I understood correctly we should move the business logic to the endpoint handler. In this case, it has to be created when the order edit is done, at the request confirmation. Is that ok?

Reply:
Yes you are right - definitely request not create sorry!